internal/attestation/snp/snp_test.go

```diff
@@ -0,0 +1,454 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package snp
+
+import (
+	"crypto/x509"
+	"encoding/hex"
+	"fmt"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/edgelesssys/constellation/v2/internal/attestation/snp/testdata"
+	"github.com/edgelesssys/constellation/v2/internal/config"
+	"github.com/edgelesssys/constellation/v2/internal/logger"
+	"github.com/google/go-sev-guest/kds"
+	"github.com/google/go-sev-guest/verify/trust"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestParseCertChain tests the parsing of the certificate chain.
+func TestParseCertChain(t *testing.T) {
+	defaultCertChain := testdata.CertChain
+	askOnly := strings.Split(string(defaultCertChain), "-----END CERTIFICATE-----")[0] + "-----END CERTIFICATE-----"
+	arkOnly := strings.Split(string(defaultCertChain), "-----END CERTIFICATE-----")[1] + "-----END CERTIFICATE-----"
+
+	testCases := map[string]struct {
+		certChain []byte
+		wantAsk   bool
+		wantArk   bool
+		wantErr   bool
+	}{
+		"success": {
+			certChain: defaultCertChain,
+			wantAsk:   true,
+			wantArk:   true,
+		},
+		"empty cert chain": {
+			certChain: []byte{},
+			wantErr:   true,
+		},
+		"more than two certificates": {
+			certChain: append(defaultCertChain, defaultCertChain...),
+			wantErr:   true,
+		},
+		"invalid certificate": {
+			certChain: []byte("invalid"),
+			wantErr:   true,
+		},
+		"ark missing": {
+			certChain: []byte(askOnly),
+			wantAsk:   true,
+		},
+		"ask missing": {
+			certChain: []byte(arkOnly),
+			wantArk:   true,
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			assert := assert.New(t)
+
+			instanceInfo := &InstanceInfo{
+				CertChain: tc.certChain,
+			}
+
+			ask, ark, err := instanceInfo.ParseCertChain()
+			if tc.wantErr {
+				assert.Error(err)
+			} else {
+				assert.NoError(err)
+				assert.Equal(tc.wantAsk, ask != nil)
+				assert.Equal(tc.wantArk, ark != nil)
+			}
+		})
+	}
+}
+
+// TestParseVCEK tests the parsing of the VCEK certificate.
+func TestParseVCEK(t *testing.T) {
```

Comment:
Since the test harness is pretty much the same, you could merge this with `TestParseCertChain`

Reply:
Not sure how you mean that. The functions do different things and have different return values. The testcases are also different so there would be no overlap.

Reply:
Ah sorry, misread the function under test